projects/kit/components/input-date-multi/input-date-multi.component.ts

```diff
@@ -132,16 +133,24 @@ export class TuiInputDateMultiComponent
 
     protected open = false;
 
-    protected readonly dateFormat = inject(TUI_DATE_FORMAT);
-    protected readonly dateSeparator = inject(TUI_DATE_SEPARATOR);
+    protected dateFormat = TUI_DEFAULT_DATE_FORMAT;
     protected readonly isMobile = inject(TUI_IS_MOBILE);
     protected readonly doneWord$ = inject(TUI_DONE_WORD);
     protected readonly filler$: Observable<string> = this.dateTexts$.pipe(
         map(dateTexts =>
-            changeDateSeparator(dateTexts[this.dateFormat], this.dateSeparator),
+            changeDateSeparator(
+                dateTexts[this.dateFormat.mode],
+                this.dateFormat.separator,
+            ),
         ),
     );
 
+    protected readonly dateFormat$ = inject(TUI_DATE_FORMAT)
```

Comment:
Now that we have DI in decorators we can actually make a decorator for this and reuse it everywhere. I'll try to implement it after we merge, should be pretty cool.
```ts
@tuiFromObservable(TUI_DATE_FORMAT)
dateFormat = TUI_DEFAULT_DATE_FORMAT
```

Reply:
cool idea